homeassistant/components/compensation/sensor.py

```diff
@@ -0,0 +1,139 @@
+"""Support for compensation sensor."""
+import logging
+
+from homeassistant.const import (
+    ATTR_ENTITY_ID,
+    ATTR_UNIT_OF_MEASUREMENT,
+    CONF_ATTRIBUTE,
+    CONF_ENTITY_ID,
+    CONF_NAME,
+    CONF_UNIT_OF_MEASUREMENT,
+    STATE_UNKNOWN,
+)
+from homeassistant.core import callback
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.event import async_track_state_change_event
+
+from .const import CONF_COMPENSATION, CONF_POLYNOMIAL, CONF_PRECISION, DATA_COMPENSATION
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_ATTRIBUTE = "attribute"
+ATTR_COEFFICIENTS = "coefficients"
+
+
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Set up the Compensation sensor."""
+    if discovery_info is not None:
```

Comment:
I'd rather return early if `discovery_info` is None.

Reply:
I tried that but then the tests would fail coverage.